crates/test-case-core/src/test_case.rs

```diff
@@ -5,7 +5,36 @@ use proc_macro2::{Span as Span2, TokenStream as TokenStream2};
 use quote::quote;
 use syn::parse::{Parse, ParseStream};
 use syn::punctuated::Punctuated;
-use syn::{parse_quote, Error, Expr, Ident, ItemFn, ReturnType, Token};
+use syn::{parse_quote, Attribute, Error, Expr, Ident, ItemFn, ReturnType, Token};
+
+// Check whether given attribute is a test attribute of forms:
+// * `#[test]`
+// * `#[core::prelude::*::test]` or `#[::core::prelude::*::test]`
+// * `#[std::prelude::*::test]` or `#[::std::prelude::*::test]`
+fn is_test_attribute(attr: &Attribute) -> bool {
+    let path = match &attr.meta {
+        syn::Meta::Path(path) => path,
+        _ => return false,
+    };
+    let candidates = [
+        ["core", "prelude", "*", "test"],
+        ["std", "prelude", "*", "test"],
+    ];
```

Comment:
This is just a code-cleanliness suggestion. You can get rid of the implicit assumption that all arrays have the same length:
```suggestion
    const CANDIDATES_LEN: usize = 4;

    let candidates: [[&str; CANDIDATES_LEN]; _] = [
        ["core", "prelude", "*", "test"],
        ["std", "prelude", "*", "test"],
    ];
```
Then you can do:
```diff
-    } else if path.segments.len() != candidates[0].len() {
+    } else if path.segments.len() != CANDIDATES_LEN {
         return false;
     }
```

Reply:
Thank you for our mental health!